### DISCUSSION TITLE
Export contacts

### DISCUSSION THREAD

**Discussion by abbasudo** [tulir/whatsmeow#175] [WhatsApp protocol Q&A] [answered]
Is there any way to export all contacts ? Or search them

**Comment by hajsf** [comment, marked as answer]
> Is there any way to export all contacts

```go
        // import "encoding/csv"

	client.FetchAppState(appstate.WAPatchCriticalUnblockLow, true, false)
	contacts, err := client.Store.Contacts.GetAllContacts()
	if err != nil {
		fmt.Println("Error reading contacts:", err)
	}
	fmt.Println("You have:", len(contacts), "contacts")

	file, err := os.Create("contacts.csv")
	if err != nil {
		log.Fatalln("failed to open file", err)
	}
	//	defer file.Close()

	w := csv.NewWriter(file)
	// defer w.Flush()

	// Using Write
	for jid, info := range contacts {
		row := []string{
			jid.User,
			info.FirstName,
			info.FullName,
			info.PushName,
			info.BusinessName,
			strconv.FormatBool(info.Found),
		}
		if err := w.Write(row); err != nil {
			log.Fatal("error writing record to file", err)
		}
	}
	w.Flush()
	file.Close()
```

**Reply by abbasudo** [reply]
> > Is there any way to export all contacts
> 
> ```go
> 	client.FetchAppState(appstate.WAPatchCriticalUnblockLow, true, false)
> 	contacts, err := client.Store.Contacts.GetAllContacts()
> 	if err != nil {
> 		fmt.Println("Error reading contacts:", err)
> 	}
> 	fmt.Println("You have:", len(contacts), "contacts")
> 
>        for _, info := range contacts {
>        // DoSomething
>         }
> ```

thanks @hajsf

**Reply by masewo** [reply]
Does anyone know why all my `FirstName` fields are empty? The first names are stored concatenated with the last names in the `FullName` fields. In the official WhatsApp app the contact information is shown separated in first name and last name correctly.
